src/main/java/org/codehaus/mojo/servicedocgen/ReportGen.java

```diff
@@ -0,0 +1,257 @@
+package org.codehaus.mojo.servicedocgen;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.ResourceBundle;
+import java.util.regex.Pattern;
+
+import javax.ws.rs.Path;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.reporting.AbstractMavenReport;
+import org.apache.maven.reporting.MavenReportException;
+import org.codehaus.mojo.servicedocgen.descriptor.ServicesDescriptor;
+import org.codehaus.mojo.servicedocgen.generation.ServicesGenerator;
+import org.codehaus.mojo.servicedocgen.generation.velocity.VelocityServicesGenerator;
+
+import com.thoughtworks.qdox.JavaProjectBuilder;
+import com.thoughtworks.qdox.model.JavaAnnotation;
+import com.thoughtworks.qdox.model.JavaClass;
+import com.thoughtworks.qdox.model.JavaSource;
+
+/**
+ * @author suvmanda
+ *
+ */
+public abstract class ReportGen extends AbstractMavenReport {
+
+  /**
+   * The constructor.
+   */
+  public ReportGen(String templateName) {
+
+    this.templateName = templateName;
+
+  }
```

Comment:
you are not using mojohaus code-style but devonfw. I know the mojohaus style sucks but this is somehow given by the project. We need to reformat this. Also nice to have and I could do that after the merge to get this included faster.